pkg/azureutils/workflow.go

```diff
@@ -0,0 +1,178 @@
+package azureutils
```

Comment:
I wonder if we should move `Workflow` and `Logger` to its own package, `workflow`. That way, creating a new workflow could be written a little more naturally as:

```go
w := workflow.New(ctx, workflow.WithDetails(...))
```

It would mean renaming all of the existing `workflow` variables, though. ☹️

Reply:
Wouldn't be a problem :D